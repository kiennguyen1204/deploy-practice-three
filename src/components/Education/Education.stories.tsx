import { StoryObj, Meta } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

// components
import { WeedEducationPattern } from 'components/Education/index';

export default {
  title: 'Components/WeedEducationPattern',
  component: WeedEducationPattern,
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    )
  ]
} as Meta;

type Story = StoryObj<typeof WeedEducationPattern>;

export const Default: Story = {};
