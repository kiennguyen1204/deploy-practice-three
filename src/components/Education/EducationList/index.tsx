import { memo } from 'react';

interface Props {
  item: {
    image: string;
    subtitle: string;
    title: string;
    description: string;
    to: string;
  };
}

const WeedEducationItem: React.FC<Props> = ({ item }): JSX.Element => {
  const { image, subtitle, title, description, to } = item;

  return (
    <article className='list-container weed-education-item' data-testid='weed-education-item'>
      <picture className='weed-education-item-image'>
        <img src={image} alt='main image' />
      </picture>

      <section className='wrapper-container weed-education-item-content'>
        <header className='wrapper-container weed-education-item-info'>
          <h2 className='weed-education-item-subtitle' data-testid='subtitle-test'>
            {subtitle}
          </h2>
          <h1 className='weed-education-item-title' data-testid='title-test'>
            {title}
          </h1>
          <p className='weed-education-item-description'>{description}</p>
        </header>
        <footer className='weed-education-item-read-more'>
          <a href={to} className='base-link'>
            Read More
          </a>
        </footer>
      </section>
    </article>
  );
};

export default memo(WeedEducationItem);
