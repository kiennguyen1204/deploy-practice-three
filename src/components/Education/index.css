.weed-education-container {
  background: var(--color-matte-white);
  padding-top: 64px;
}

.weed-education-container .card-box {
  padding: 32px 24px;
}

.weed-education-container .card-box .card-box-element {
  background: var(--color-matte-white);
  padding: 0 0 64px 0;
}

.weed-education-container .card-box .card-box-element .weed-education-group {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: flex-start;
  gap: 24px;
  width: 100%;
}

.weed-education-container .card-box .card-box-element .weed-education-group .weed-education-item {
  justify-content: flex-start;
  align-items: stretch;
  gap: 24px;
  position: relative;
  max-width: 100%;
  width: 100%;
  height: auto;
}

.weed-education-container
  .card-box
  .card-box-element
  .weed-education-group
  .weed-education-item
  .weed-education-item-image
  img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 8px;
  image-rendering: auto;
}

.weed-education-container
  .card-box
  .card-box-element
  .weed-education-group
  .weed-education-item
  .weed-education-item-content {
  gap: 32px;
}

.weed-education-container
  .card-box
  .card-box-element
  .weed-education-group
  .weed-education-item
  .weed-education-item-content
  .weed-education-item-info {
  gap: 8px;
}

.weed-education-container
  .card-box
  .card-box-element
  .weed-education-group
  .weed-education-item
  .weed-education-item-content
  .weed-education-item-info
  .weed-education-item-subtitle {
  color: var(--color-neutral-light);
  font-size: var(--font-size-normal);
  font-weight: var(--font-weight-light);
  line-height: var(--line-height-small);
}

.weed-education-container
  .card-box
  .card-box-element
  .weed-education-group
  .weed-education-item
  .weed-education-item-content
  .weed-education-item-info
  .weed-education-item-title {
  color: var(--color-neutral-black);
  font-size: var(--font-size-xxl);
  font-weight: var(--font-weight-medium);
  line-height: var(--line-height-xl);
  letter-spacing: -0.5px;
}

.weed-education-container
  .card-box
  .card-box-element
  .weed-education-group
  .weed-education-item
  .weed-education-item-content
  .weed-education-item-info
  .weed-education-item-description {
  color: var(--color-neutral-light);
  font-size: var(--font-size-normal);
  font-weight: var(--font-weight-medium);
  line-height: var(--line-height-medium);
}

.weed-education-container
  .card-box
  .card-box-element
  .weed-education-group
  .weed-education-item
  .weed-education-item-content
  .weed-education-item-read-more
  .base-link {
  color: var(--color-primary-base);
  font-size: var(--font-size-normal);
  font-weight: var(--font-weight-regular);
  line-height: var(--line-height-medium);
  text-decoration: underline;
}

@media screen and (min-width: 768px) {
  .weed-education-container .card-box .card-box-element .weed-education-group {
    gap: 32px;
    justify-content: space-between;
  }

  .weed-education-container .card-box .card-box-element .weed-education-group .weed-education-item {
    max-width: calc(100% / 2 - 24px);
  }
}

@media screen and (min-width: 992px) {
  .weed-education-container .card-box .card-box-element .weed-education-group .weed-education-item {
    max-width: calc(100% / 3 - 24px);
  }
}

@media screen and (min-width: 1200px) {
  .weed-education-container {
    padding-top: 120px;
  }

  .weed-education-container .card-box {
    padding: 0 120px;
  }
}
