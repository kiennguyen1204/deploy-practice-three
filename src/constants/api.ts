export const URL_API = {
  BASE_URL: process.env.VITE_BASE_URL,
  PRODUCTS_URL: '/products'
};

const PRODUCTS_API = `${URL_API.BASE_URL}${URL_API.PRODUCTS_URL}`;

export { PRODUCTS_API };
