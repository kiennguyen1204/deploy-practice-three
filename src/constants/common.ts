export const MAX_STAR = 5;
