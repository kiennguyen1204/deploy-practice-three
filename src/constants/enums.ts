export enum BUTTON_VARIANT {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  TERTIARY = 'tertiary'
}

export enum BUTTON_SIZE {
  SMALL = 'sm',
  MEDIUM = 'md',
  LARGE = 'lg',
  EXTRA_LARGE = 'xl'
}

export enum BUTTON_COLOR {
  SUCCESS = 'success',
  WARNING = 'warning',
  DEFAULT = 'default'
}

export enum BUTTON_COLOR_TEXT {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  TERTIARY = 'tertiary'
}

export enum BUTTON_TYPE {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset'
}

export enum INPUT_VARIANT {
  DEFAULT = 'default',
  PRIMARY = 'primary'
}

export enum INPUT_TYPE {
  TEXT = 'text',
  NUMBER = 'number'
}

export enum INPUT_SIZE {
  SMALL = 'sm',
  MEDIUM = 'md',
  LARGE = 'lg',
  EXTRA_LARGE = 'xl'
}

export enum STATUS {
  SUCCESS = 'success',
  ERROR = 'error'
}

export enum TYPE_LINKS {
  QUICK = 'quick',
  MORE = 'more',
  PAYMENT = 'payment'
}

export enum OPTION_WEED {
  FLOWERS,
  MUSHROOMS,
  CONCENTRACE
}

export enum STEPS_CART {
  SHOPPING_CART,
  CHECKOUT,
  ORDER_COMPLETE
}

export enum SORT_TYPE {
  ASC = 'price_asc',
  DESC = 'price_desc'
}