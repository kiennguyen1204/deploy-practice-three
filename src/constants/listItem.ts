export const LIST_ITEM_NAVIGATION = [
  { name: 'Shop All', to: '/about' },
  {
    name: 'Flower',
    to: ''
  },
  { name: 'Edibles', to: '' },
  {
    name: 'Concentrates',
    to: ''
  },
  { name: 'Mushrooms', to: '' },
  {
    name: 'Promotions/Bundles',
    to: ''
  },
  {
    name: 'Support',
    to: ''
  },
  { name: 'Rewards', to: '' },
  { name: 'Blog', to: '' }
];

export const DATA_INFO = [
  {
    title: 'Indica',
    image: '/src/assets/images/indica.png',
    icon: '/src/assets/images/sleeping_face.svg',
    description: `This type of cannabis is commonly taken by those who want to sink into a state of total relaxation in every limb. This reduces stress overall and can take your worries and fatigue away. Because of its relaxing effects, it is suggested to use this type of cannabis at night. It is particularly recommended for people who have trouble sleeping, be it due to insomnia, pain or other associated sleep issues.`,
    to: '/indica',
    shop: 'Shop Indica'
  },
  {
    title: 'Sativa',
    image: '/src/assets/images/indica.png',
    icon: '/src/assets/images/sun.svg',
    description:
      'Contrary to the deep all-body relaxation that comes with the use of indica strains, sativas are known for increasing the feeling of creativity, enhancing focus and lessening anxiety. This is more of a mind-centered high in terms of how and where you will feel the effects. Given its stimulating nature, it is recommended to use this during the day.',
    to: '/sativa',
    shop: 'Shop Sativa'
  },
  {
    title: 'Hybrids',
    image: '/src/assets/images/hybrids.png',
    description: `Hybrid strains are just that – they combine the head-focused high effects of sativas with the bodily relaxation of indicas. This is ideal for people who really want to sooth any fatigue and worries while also clearing the mind ready to focus fresh on something new. Hybrids are expertly tailored to result in strains with specific effects. Useful strain vocabulary to familiarize yourself with when it comes to hybrids includes sativa-dominant (full, bodily relaxation), indica-dominant (boosting creativity, increasing mood and lessening anxiety) and balanced (the best of both worlds).`,
    top: '/hybrids',
    shop: 'Shop Hybrids'
  }
];

export const LIST_ITEM_EDUCATION = [
  {
    image: '/src/assets/images/education/info.webp',
    title: '12 Mistakes To Avoid When Buying Cannabis Online',
    subtitle: 'January 24, 2023',
    description:
      'Buying cannabis online can be a great way to get your hands on the products you need without leaving the comfort of your home. But …',
    to: '/home'
  },
  {
    image: '/src/assets/images/education/weight.webp',
    title: 'How To Store Cannabis and Keep it Fresh and Potent?',
    subtitle: 'January 20, 2023',
    description:
      'Cannabis packaging has advanced dramatically in recent years, whether your state has a medicinal marijuana programme, legal adult-use weed, or both. Most ...',
    to: '/home'
  },
  {
    image: '/src/assets/images/education/weight.webp',
    title: 'The Ultimate Guide to Checking the Quality of Cannabis – 10 Industry Leading Tips',
    subtitle: 'January 19, 2023',
    description:
      'Quality cannabis is a term used to describe cannabis products that meet specific standards of excellence. It is essential to understand what quality cannabis means, …',
    to: '/home'
  }
];
