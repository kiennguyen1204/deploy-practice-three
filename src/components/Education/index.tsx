// constants
import { LIST_ITEM_EDUCATION } from 'constants/listItem';

// components
import Card from 'components/common/Card';
import WeedEducationItem from './EducationList';

// styles
import './index.css';

export const WeedEducationPattern: React.FC = (): JSX.Element => (
  <Card classes={'weed-education-container'} title='WEED EDUCATION' iShowAll>
    <main className='weed-education-box'>
      <section className='weed-education-group'>
        {LIST_ITEM_EDUCATION.map((item, index) => (
          <WeedEducationItem key={index} item={item} />
        ))}
      </section>
    </main>
  </Card>
);
