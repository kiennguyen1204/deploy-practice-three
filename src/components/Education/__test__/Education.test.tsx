import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

// constants
import { LIST_ITEM_EDUCATION } from 'constants/listItem';

// components
import { WeedEducationPattern } from 'components/Education/index';

describe('WeedEducationPattern component', () => {
  it('renders correctly', () => {
    const { getByText, getAllByTestId, asFragment } = render(
      <MemoryRouter>
        <WeedEducationPattern />
      </MemoryRouter>
    );

    const titleElement = getByText('WEED EDUCATION');
    expect(titleElement).toBeInTheDocument();

    const showAllButton = getByText('Show All');
    expect(showAllButton).toBeInTheDocument();

    const weedEducationItems = getAllByTestId('weed-education-item');
    expect(weedEducationItems.length).toBe(LIST_ITEM_EDUCATION.length);

    expect(asFragment()).toMatchSnapshot();
  });
});
