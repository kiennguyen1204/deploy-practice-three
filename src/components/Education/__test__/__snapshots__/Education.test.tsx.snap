// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeedEducationPattern component renders correctly 1`] = `
<DocumentFragment>
  <article
    class="card-box-container weed-education-container"
  >
    <header
      class="flex-space-between-container header-title"
    >
      <h2
        class="title"
      >
        WEED EDUCATION
      </h2>
      <a
        class="show-all-link"
        href="https://weedmaps.com/learn"
      >
        Show All
      </a>
    </header>
    <section
      class="card-box"
    >
      <div
        class="card-box-element"
      >
        <main
          class="flex-start-container weed-education-box"
        >
          <section
            class="weed-education-group"
          >
            <article
              class="list-container weed-education-item"
              data-testid="weed-education-item"
            >
              <picture
                class="weed-education-item-image"
              >
                <img
                  alt="main image"
                  src="/src/assets/images/education/info.webp"
                />
              </picture>
              <section
                class="wrapper-container weed-education-item-content"
              >
                <header
                  class="wrapper-container weed-education-item-info"
                >
                  <h2
                    class="weed-education-item-subtitle"
                    data-testid="subtitle-test"
                  >
                    January 24, 2023
                  </h2>
                  <h1
                    class="weed-education-item-title"
                    data-testid="title-test"
                  >
                    12 Mistakes To Avoid When Buying Cannabis Online
                  </h1>
                  <p
                    class="weed-education-item-description"
                  >
                    Buying cannabis online can be a great way to get your hands on the products you need without leaving the comfort of your home. But …
                  </p>
                </header>
                <footer
                  class="weed-education-item-read-more"
                >
                  <a
                    class="base-link"
                    href="/home"
                  >
                    Read More
                  </a>
                </footer>
              </section>
            </article>
            <article
              class="list-container weed-education-item"
              data-testid="weed-education-item"
            >
              <picture
                class="weed-education-item-image"
              >
                <img
                  alt="main image"
                  src="/src/assets/images/education/weight.webp"
                />
              </picture>
              <section
                class="wrapper-container weed-education-item-content"
              >
                <header
                  class="wrapper-container weed-education-item-info"
                >
                  <h2
                    class="weed-education-item-subtitle"
                    data-testid="subtitle-test"
                  >
                    January 20, 2023
                  </h2>
                  <h1
                    class="weed-education-item-title"
                    data-testid="title-test"
                  >
                    How To Store Cannabis and Keep it Fresh and Potent?
                  </h1>
                  <p
                    class="weed-education-item-description"
                  >
                    Cannabis packaging has advanced dramatically in recent years, whether your state has a medicinal marijuana programme, legal adult-use weed, or both. Most ...
                  </p>
                </header>
                <footer
                  class="weed-education-item-read-more"
                >
                  <a
                    class="base-link"
                    href="/home"
                  >
                    Read More
                  </a>
                </footer>
              </section>
            </article>
            <article
              class="list-container weed-education-item"
              data-testid="weed-education-item"
            >
              <picture
                class="weed-education-item-image"
              >
                <img
                  alt="main image"
                  src="/src/assets/images/education/weight.webp"
                />
              </picture>
              <section
                class="wrapper-container weed-education-item-content"
              >
                <header
                  class="wrapper-container weed-education-item-info"
                >
                  <h2
                    class="weed-education-item-subtitle"
                    data-testid="subtitle-test"
                  >
                    January 19, 2023
                  </h2>
                  <h1
                    class="weed-education-item-title"
                    data-testid="title-test"
                  >
                    The Ultimate Guide to Checking the Quality of Cannabis – 10 Industry Leading Tips
                  </h1>
                  <p
                    class="weed-education-item-description"
                  >
                    Quality cannabis is a term used to describe cannabis products that meet specific standards of excellence. It is essential to understand what quality cannabis means, …
                  </p>
                </header>
                <footer
                  class="weed-education-item-read-more"
                >
                  <a
                    class="base-link"
                    href="/home"
                  >
                    Read More
                  </a>
                </footer>
              </section>
            </article>
          </section>
        </main>
      </div>
    </section>
  </article>
</DocumentFragment>
`;
